### PR TITLE
feat: Add animation classes.

### PR DESCRIPTION
**Pull request by tjameswhite**
**Related Issue:** #2411 

## Summary
Animation keyframes exist in Calcite, but they are not available via any classes. 

Add `.calcite-animate` and related classes
Add `:root {--calcite-animation-timing: 0.6s;}` 

Remainder of implementation is up to end user.

### CODE REVIEW

**Review by driskull — APPROVED** [review]
This looks good to me. @julio8a @asangma what do you think?

**Comment by julio8a** [comment]
@macandcheese, can you take a look?

**Comment by macandcheese** [comment]
I don’t mind the idea of opt in classes, I’ll  defer to @bstifle on the actual transitions and speeds.

I guess my only thought is that this would perhaps sit better adjacent to the tailwind classes generated in the calcite-styles repo than here. That’s still not available so maybe here is fine in the interim, but as of now no components use these internally - I think long term these belong elsewhere and CC can just consume them. That way folks can use these as they would a design token or tw class.

**Comment by driskull** [comment]
>  perhaps sit better adjacent to the tailwind classes

Can we move these to tailwind? That would make more sense if we have a tokens repo eventually.

**Comment by macandcheese** [comment]
> > perhaps sit better adjacent to the tailwind classes
> 
> Can we move these to tailwind? That would make more sense if we have a tokens repo eventually.

Yeah that makes more sense to me. Either in the tailwind config (that eventually gets generated as class names for folks to use w/o CC), or in that tokens / styles repo.

**Comment by tjameswhite** [comment]
For what its worth -- timing was set by @cebender for our needs on esri.com. Most components in AEM have an animation on them, so that is our use case. We need to complete an audit of current animations, but the plan is to move to CC animation soonish. (I was actually looking to other css animation libraries when I found this already in CC.) One reason I used the css custom property for timing was to provide flexibility if others want to adjust.

**Comment by caripizza** [comment]
Hi @tjameswhite - just following up from my Teams message. Julio clarified to me yesterday that yes, we would like to add this work to calcite-components. (It will be some time before we get the Tailwind npm package ready to be consumed in calcite-components, so this is a better home for the time being.)

@julio8a can you add Tim to the [GitHub team](https://github.com/orgs/Esri/teams/calcite-components/members), so he can clone the repo (non-fork) and open a new pr?

**Comment by julio8a** [comment]
Tim has write access to the repo.

**Comment by caripizza** [comment]
Closing in favor of pr #2607

**Pull request closed without merging**
